### DISCUSSION TITLE
Why does the JSON schema for 3.0 specify patternProperties for component names but allows additional properties?

### DISCUSSION THREAD

**Discussion by ryankinderman** [OAI/OpenAPI-Specification#2439] [Q&A] [answered]
I've been working on some functionality that uses the JSON schema to help validate OpenAPI schemas, and recently noticed that the 3.0 schema allows any value for `{name}` in `/components/{component}/{name}`, while the [specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#fixed-fields-6) states that "All the fixed fields declared above are objects that MUST use keys that match the regular expression: `^[a-zA-Z0-9\.\-_]+$.`" It seems like this could have been validated automatically by simply adding `additionalProperties: false` to the JSON schema.

My question is: Was not including `additionalProperties: false` in the JSON schema for 3.0 just an accidental omission, or was there some rationale behind not doing it?

Thanks for clarifying!

**Comment by MikeRalphson** [comment]
I *think* this does look like an accidental oversight. Perhaps @webron could confirm? 

Would you be interested in raising a PR against the (YAML version of the) schema to add the missing keywords? We will then regenerate the JSON version and bump the `id` value as a separate commit.

**Reply by ryankinderman** [reply]
I’d be happy to submit a PR, but that seems like it may be a breaking change for people currently using the 3.0 JSON schema for structural validation as it’d cause it to start failing where previously it wasn’t. Not sure how that’d affect how this gets addressed.

**Comment by handrews** [comment, marked as answer]
Migrated to issue #3720 as there's a clear potential action.  Closing this discussion in favor of the issue.
